tests/infra/clients.py

```diff
@@ -366,33 +367,50 @@ def request(self, request, is_signed=False):
 
 class WSClient:
     def __init__(
-        self,
-        host,
-        port,
-        cert,
-        key,
-        ca,
-        connection_timeout,
-        request_timeout,
-        *args,
-        **kwargs,
+        self, host, port, cert, key, ca, request_timeout, *args, **kwargs,
     ):
         self.host = host
         self.port = port
         self.cert = cert
         self.key = key
         self.ca = ca
         self.request_timeout = request_timeout
+        self.ws = None
 
-    def request(self, request):
-        # pylint: disable=unused-variable
-        ws = create_connection(
-            f"wss://{self.host}:{self.port}",
-            sslopt={"certfile": self.cert, "keyfile": self.key, "ca_certs": self.ca},
+    def request(self, request, is_signed=False):
+        assert not is_signed
+
+        if not self.ws:
+            LOG.info("Creating WSS connection")
+            self.ws = websocket.create_connection(
+                f"wss://{self.host}:{self.port}",
+                sslopt={
+                    "certfile": self.cert,
+                    "keyfile": self.key,
+                    "ca_certs": self.ca,
+                },
+            )
+        payload = json.dumps(request.params).encode()
+        path = ("/" + request.method).encode()
+        header = struct.pack("<h", len(path)) + path
+        # FIN, no RSV, BIN, UNMASKED every time, because it's all we support right now
+        frame = websocket.ABNF(
+            1, 0, 0, 0, websocket.ABNF.OPCODE_BINARY, 0, header + payload
         )
-
-    def signed_request(self, request):
-        raise NotImplementedError("Signed requests not yet implemented over WebSockets")
+        self.ws.send_frame(frame)
+        out = self.ws.recv_frame().data
```

Comment:
And a `TimeoutError` here if we don't receive anything.

Reply:
We get that from websocket, but we should set the timeout!